numpy/lib/_function_base_impl.py

```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
 
-    closest_observation:
-        method 3 of H&F [1]_.
-        This method gives discontinuous results:
+        (1-g)*y[j] + g*y[j+1]
 
-        * if g > 0 ; then take j
-        * if g = 0 and index is odd ; then take j
-        * if g = 0 and index is even ; then take i
+    where the index ``j`` and coefficient ``g`` are the integral and
+    fractional components of ``q * (n-1)``, and ``n`` is the number of
+    elements in the sample.
 
-    interpolated_inverted_cdf:
-        method 4 of H&F [1]_.
-        This method gives continuous results using:
+    This a special case of Equation 1 of H&F [1]_. More generally,
 
-        * alpha = 0
-        * beta = 1
+    - ``j = (q*n + m - 1) // 1``, and
+    - ``g = (q*n + m - 1) % 1``,
 
-    hazen:
-        method 5 of H&F [1]_.
-        This method gives continuous results using:
-
-        * alpha = 1/2
-        * beta = 1/2
-
-    weibull:
-        method 6 of H&F [1]_.
-        This method gives continuous results using:
-
-        * alpha = 0
-        * beta = 0
-
-    linear:
-        method 7 of H&F [1]_.
-        This method gives continuous results using:
-
-        * alpha = 1
-        * beta = 1
-
-    median_unbiased:
-        method 8 of H&F [1]_.
-        This method is probably the best method if the sample
-        distribution function is unknown (see reference).
-        This method gives continuous results using:
-
-        * alpha = 1/3
-        * beta = 1/3
-
-    normal_unbiased:
-        method 9 of H&F [1]_.
-        This method is probably the best method if the sample
-        distribution function is known to be normal.
-        This method gives continuous results using:
-
-        * alpha = 3/8
-        * beta = 3/8
-
-    lower:
-        NumPy method kept for backwards compatibility.
-        Takes ``i`` as the interpolation point.
-
-    higher:
-        NumPy method kept for backwards compatibility.
-        Takes ``j`` as the interpolation point.
-
-    nearest:
-        NumPy method kept for backwards compatibility.
-        Takes ``i`` or ``j``, whichever is nearest.
+    where ``m`` is defined differently for each value of the ``method``
+    parameter.
 
-    midpoint:
-        NumPy method kept for backwards compatibility.
-        Uses ``(i + j) / 2``.
+    +-------------------------------+---------------+---------------+
+    | ``method``                    | # in H&F [1]_ | ``m``         |
+    +===============================+===============+===============+
+    | ``interpolated_inverted_cdf`` | 4             | ``0``         |
+    | ``hazen``                     | 5             | ``1/2``       |
+    | ``weibull``                   | 6             | ``q``         |
+    | ``linear`` (default)          | 7             | ``1 - q``     |
+    | ``median_unbiased``           | 8             | ``q/3 + 1/3`` |
+    | ``normal_unbiased``           | 9             | ``q/4 + 3/8`` |
+    +-------------------------------+---------------+---------------+
```

Comment:
This table is not showing up in the rendered output. Any suggestions?

Reply:
```
/home/circleci/repo/venv/lib/python3.11/site-packages/numpy/__init__.py:docstring of numpy.quantile:144: ERROR: Malformed table.

+-------------------------------+---------------+---------------+
| ``method``                    | # in H&F [Re01cd3f3acfe-1]_ | ``m``         |
+===============================+===============+===============+
| ``interpolated_inverted_cdf`` | 4             | ``0``         |
| ``hazen``                     | 5             | ``1/2``       |
| ``weibull``                   | 6             | ``q``         |
| ``linear`` (default)          | 7             | ``1 - q``     |
| ``median_unbiased``           | 8             | ``q/3 + 1/3`` |
| ``normal_unbiased``           | 9             | ``q/4 + 3/8`` |
+-------------------------------+---------------+---------------+
```
Is it the reference being auto-expanded by numpy-doc maybe?  Or does sphinx not like the `=`?  Not sure.

Reply:
I'll try taking out the reference. That's probably it. Thanks! I saw that CircleCI was green and didn't look closer.

Reply:
Hmm that warning is gone but it's still not rendering appropriately.

<img width="582" alt="image" src="https://github.com/numpy/numpy/assets/6570539/c6a0e67a-2dde-47f3-a48d-87cd7a4803cd">

Reply:
I saw you had a try with using `===== ===== ====` style table?  That seeme dlike a good idea to me:  It is what the sort kind table uses as well.